### PR TITLE
EDU-3093: Updates user page to add Finance Role

### PR DESCRIPTION
**Pull request by fairlydurable**
* Updates role listings
* Updates table
* Adds permission of account owner to create users

Known omissions: There will be additional table entries specific to finance actions that must be added. UPDATE: added.

### CODE REVIEW

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 1]
```diff
@@ -52,7 +52,7 @@ However, those using Google or Microsoft for SSO authentication need to follow t
 
 :::info
 
-To invite users, a user must have the Global Admin account-level [Role](#account-level-roles).
+To invite users, a user must have the Global Admin or Account Owner account-level [Role](#account-level-roles).
```
Comment (angelazhou32) [inline docs/production-deployment/cloud/account-setup/users.mdx:55]: Is there a reason the role needs to be categorized?

```suggestion
To invite users, a user must have the Global Admin or Account Owner account-level [role](#account-level-roles).
```

File: docs/production-deployment/cloud/account-setup/users.mdx [hunk 2]
```diff
@@ -109,11 +109,14 @@ When a Global Admin invites a user to join an account, the Global Admin selects
   - Has Namespace Admin permissions for each Namespace created by that user
 - **Read-Only:** Can only read information
 
-In addition there's one role that the Global Admin cannot assign:
+In addition there are two roles that the Global Admin cannot assign:
```
Comment (angelazhou32) [inline docs/production-deployment/cloud/account-setup/users.mdx:112]:
```suggestion
In addition, there are two roles that the Global Admin cannot assign:
```

File: docs/production-deployment/cloud/billing-and-cost.mdx [hunk 3]
```diff
@@ -30,7 +30,7 @@ tags:
   - invoice history
 ---
 
-The Account Owner holds the ability to access and view detailed billing and cost information at any time, ensuring full transparency and control over financial aspects. If you need to assess your spending patterns, check invoice histories, or handle payment details, Temporal Cloud's Cost and Billing capabilities are able to provide the information you need.
+The Account Owner and Finance Admin hold the ability to access and view detailed billing and cost information at any time, ensuring full transparency and control over financial aspects. If you need to assess your spending patterns, check invoice histories, or handle payment details, Temporal Cloud's Cost and Billing capabilities are able to provide the information you need.
```
Comment (angelazhou32) [inline docs/production-deployment/cloud/billing-and-cost.mdx:33]: Do you want to link to the billing and cost page? https://docs.temporal.io/cloud/billing-and-cost

File: docs/production-deployment/cloud/account-setup/users.mdx (hunk 2 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/account-setup/users.mdx:112]: Thank you!

File: docs/production-deployment/cloud/billing-and-cost.mdx (hunk 3 above)
Comment (fairlydurable) [reply docs/production-deployment/cloud/billing-and-cost.mdx:33]: They didn't ask for this, but I think that would be a great addition. Adding.

**Comment by fairlydurable** [comment]
Did a global audit of uppercase Role across document.

**Review by angelazhou32 — APPROVED** [review]

**Pull request merged**